# VGP parameter optimization


## Generation of  synthetic PacBio HiFi reads

The synthetic Pacbio HiFi reads were generated by using the *Saccharomyces cerevisiae* [reference genome](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.fna.gz) from NCBI and a subset 30X of the dataset SRR13577847.
    
```
$ ./FastK -T8 -Nsynthetic -k40 -t4 ./hifi_reads.fastq.gz
$ ./FastK -T8 -Nsynthetic -k40 -p:synthetic.ktab ../inputs/hifi_reads.fastq.gz
$ ./Histex -G synthetic > synthetic.histogram
$ ./GeneScopeFK.R -i synthetic.histogram -o ./synthetic -k 40 -p 1
$ ./HImodel -v -osynthetic -g10:55 -e5 -T8 ./synthetic
$ ./HIsim ./GCF_000146045.2_R64_genomic.fasta ./synthetic.model -p.2,.2 -c50 -r1 -U -f -h -e
```
    
## Workflow details:

### Full purge_dups pipeline

It includes the [full post-assembly pipeline](https://training.galaxyproject.org/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html#post-assembly-processing): read-depth analysis, generation of all versus all self-alignment and resolution of haplotigs and overlaps. In addition, it runs QUAST and Merqury both at the beginning and the end.

- Workflow ID: 08e65ff9a4f1ebb1
    
- Input files:
  - Pacbio HiFi reads
  - Draft assembly
  - K-mer database

### Partial purge_dups pipeline

In includes only the thid stage of the purge_dups pipeline (resolution of haplotics and overlaps) and a final report by running QUAST and Merqury.

- Workflow ID: 237014729f9f1b35
- Input files:
  - Draft assembly
  - K-mer database
  - Pacbio HiFi reads
  - Calcuts report
  - PBCSTAT report
  - Self-homologous map
